internal/host/plugin/repository_host_catalog.go

```diff
@@ -143,6 +146,279 @@ func (r *Repository) CreateCatalog(ctx context.Context, c *HostCatalog, _ ...Opt
+// values in c for the fields listed in fieldMask. It returns a new
+// HostCatalog containing the updated values and a count of the number of
+// records updated. c is not changed.
+//
+// c must contain a valid PublicId. c.Name, c.Description, and
+// c.Attributes can be updated; if c.Secrets is present, its contents
+// are sent to the plugin (along with any other changes, see below)
+// before the update is sent to the database.
+//
+// An attribute of c will be set to NULL in the database if the
+// attribute in c is the zero value and it is included in fieldMask.
+// Note that this does not apply to c.Attributes - a null
+// c.Attributes is a no-op for modifications. Rather, if fields need
+// to be reset, its field in c.Attributes should individually set to
+// null.
+//
+// Updates are sent to OnUpdateCatalog with a full copy of both the
+// current catalog, and the state of the new catalog should it be
+// updated, along with any secrets included in the new request. This
+// request may alter the returned persisted state. Update of the
+// record in the database is aborted if this call fails.
+//
+// The first integer value represents the number of catalogs updated.
+// The second represents the number of secrets updated. There should
+// always be one catalog updated if there is no error. It's possible
+// that zero rows may be affected for secrets, but this will be rare,
+// as this operation is opaque and the update is only skipped if
+// there are no secrets returned altogether.
+func (r *Repository) UpdateCatalog(ctx context.Context, c *HostCatalog, version uint32, fieldMask []string, _ ...Option) (*HostCatalog, int, int, error) {
+	const op = "plugin.(Repository).UpdateCatalog"
+	if c == nil {
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.New(ctx, errors.InvalidParameter, op, "nil HostCatalog")
+	}
+	if c.HostCatalog == nil {
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.New(ctx, errors.InvalidParameter, op, "nil embedded HostCatalog")
+	}
+	if c.PublicId == "" {
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.New(ctx, errors.InvalidParameter, op, "no public id")
+	}
+	if c.ScopeId == "" {
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.New(ctx, errors.InvalidParameter, op, "no scope id")
+	}
+	if len(fieldMask) == 0 {
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.New(ctx, errors.EmptyFieldMask, op, "empty field mask")
+	}
+
+	// Quickly replace the passed in HostCatalog with a clone. This
+	// ensures that we don't alter anything in the original passed in
+	// parameters, which we don't do by convention. We will be adding
+	// to this "working set" as we move through the method.
+	c = c.clone()
+
+	// Get the old catalog first. We patch the record first before
+	// sending it to the DB for updating so that we can run on
+	// OnUpdateCatalog. Note that the field masks are still used for
+	// updating.
+	currentCatalog, _, err := r.LookupCatalog(ctx, c.PublicId)
```

Comment:
use getCatalog instead of LookupCatalog.

Reply:
Yep, updated on merge yesterday.